src/Behat/Gherkin/Parser.php

```diff
@@ -635,6 +638,26 @@ protected function popTags()
         return $tags;
     }
 
+    /**
+     * Checks the tags fit the required format
+     *
+     * @param string[] $tags
+     */
+    protected function guardTags(array $tags)
+    {
+        foreach ($tags as $tag) {
+            if (preg_match('/\s/', $tag)) {
+                throw new ParserException(
```

Comment:
This could be `E_USER_DEPRECATED` if we think whitespace tags are a valid current usage rather than an error

Reply:
What do you think about this @stof?

Reply:
do tag expressions in TagFilter support tags with whitespace in them or no ? If such tags are not usable in TagFilter, then I would go for a ParserException